src/pcrlock/pcrlock.c

```diff
@@ -777,6 +777,16 @@ static int event_log_record_extract_firmware_description(EventLogRecord *rec) {
                 const packed_EFI_DEVICE_PATH *dp = (const packed_EFI_DEVICE_PATH*) load->devicePath;
                 size_t left = load->lengthOfDevicePath;
 
+                /* device path could be empty. Don't mark that as invalid but leave as don't know.
+                 * Happens eg with shim https://github.com/rhboot/shim/issues/642 */
+                if (left == 0) {
+                        rec->description = strdup("File:");
```

Comment:
hmm, I'd use a more explicit string, maybe `"File: <unspecified>"`?

Reply:
hmm, I'd move this three lines up with, and check `load->lengthOfDevicePath` directly instead of `left`. There's no need to prepare left/dp unless we actually want to iterate through the loop.